extends CharacterBody2D

@onready var AI = preload("res://UIPopup/UIPopup.tscn")
@onready var anim = $AnimationPlayer
@onready var player = get_tree().root.get_node("PlayerX")
@onready var ekey = $eKey
@onready var inventoryPopup = preload("res://HUD/InventorySelect.tscn")
var inventoryInstance
var aiInstance
var interacting = false

# var for slots and inventory
var slotsSelected


# Called when the node enters the scene tree for the first time.
func _ready():
	anim.speed_scale = 0.5
	anim.play("idle")
	ekey.visible = false


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	if player.position.distance_to(self.position) < 50:
		ekey.visible = true
		if Input.is_action_just_pressed("interact") and !interacting:
			interacting = true
			inventoryInstance = inventoryPopup.instantiate()
			get_parent().add_child(inventoryInstance)
			inventoryInstance.sellItems.connect(soldItems)
			#get_tree().get_root().add_child(AI.instantiate())
	else:
		ekey.visible = false

func soldItems(x):
	var totalSold = 0
	print("Selling: ", x) # where x is the dictionary that  
	for i in x.values():
		totalSold += i
	print("Total Sold: ", totalSold)

	aiInstance = AI.instantiate()
	get_tree().get_root().add_child(aiInstance)

	# CONNECT SIGNAL FOR DEAL ENDING
	aiInstance.get_child(0).dealEnded.connect(_on_dealEnded)

	aiInstance.get_child(0).initPopup("Old Man", "Player", "Item is usually worth: "+str(totalSold)+"\n Hello sir!")

	# FREE THE INVENTORY OF SOLD ITEMS
	inventoryInstance.queue_free()
	player.process_mode = PROCESS_MODE_DISABLED
	slotsSelected = x

func _on_dealEnded():
	print("xxxxxxx deal ended !!")
	for i in slotsSelected:
		print(i)
		player.get_node("Inventory").get_node("Slots").get_node(i).get_child(1).queue_free()
	aiInstance.queue_free()
	player.process_mode = PROCESS_MODE_ALWAYS

	